test/integ/pandas/test_pandas_tools.py

```diff
@@ -340,10 +351,44 @@ def test_special_name_quoting(
             # Check table's contents
             result = cnx.cursor(DictCursor).execute(select_sql).fetchall()
             for row in result:
-                assert row["id"] in (1, 2)
+                # The auto create table functionality does not auto-create an incrementing ID
+                if not auto_create_table:
+                    assert row["id"] in (1, 2)
                 assert (
                     row["00name"],
                     row["bAlance"],
                 ) in df_data
         finally:
             cnx.execute_string(drop_sql)
+
+
+def test_auto_create_table_similar_column_names(
+    conn_cnx: Callable[..., Generator["SnowflakeConnection", None, None]],
+):
+    """Tests whether similar names cause issues when auto-creating a table as expected."""
+    table_name = "numbas"
+    df_data = [(10, 11), (20, 21)]
+
+    df = pandas.DataFrame(df_data, columns=["number", "Number"])
+    select_sql = f'SELECT * FROM "{table_name}"'
+    drop_sql = f'DROP TABLE IF EXISTS "{table_name}"'
+    with conn_cnx() as cnx:  # type: SnowflakeConnection
```

Comment:
You shouldn't need that type annotation
```suggestion
    with conn_cnx() as cnx:
```